kamon-play-2.6.x/src/main/scala/kamon/play/instrumentation/RequestHandlerInstrumentation.scala

```diff
@@ -15,48 +15,54 @@
 
 package kamon.play.instrumentation
 
-import akka.http.scaladsl.model.{HttpRequest, HttpResponse}
 import kamon.Kamon
 import kamon.context.Context
-import kamon.play.OperationNameFilter
 import kamon.trace.Span
 import kamon.util.CallingThreadExecutionContext
-import org.aspectj.lang.ProceedingJoinPoint
-import org.aspectj.lang.annotation._
-import play.api.mvc.EssentialFilter
 
 import scala.concurrent.Future
 
-@Aspect
-class RequestHandlerInstrumentation {
+trait GenericRequest {
+  val headers: Map[String, String]
+  val method: String
+  val url: String
+  val spanKind: String
+}
+
+trait GenericResponse {
```

Comment:
This type might need some sort of "write" side. One thing we are doing on the Akka HTTP module is writing the context in responses which is specially useful when testing so that at least the trace ID can be know to check logs/zipkin/xxx. That is a bit tricky with immutable models, though :(.. WDYT about the feature and a approach to do it here as well?

Reply:
I'm not sure to always return the headers even if only for the purpose of testing, maybe with a flag or configuration would be fine, and the only module that does that is akka-http would also have to check kamon-http4s.